src/safeds/exceptions/_generic.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+
+class OutOfBoundsError(ValueError):
+    """A generic exception that can be used to signal that a (float) value is outside its expected range."""
+
+    def __init__(self, actual: float, *, lower_bound: Bound | None = None, upper_bound: Bound | None = None):
+        """
+        Initialize an OutOfBoundsError.
+
+        Parameters
+        ----------
+        actual: float
+            The actual value that is outside its expected range.
+        lower_bound: Bound | None
+            The lower bound of the expected range.
+        upper_bound: Bound | None
+            The upper bound of the expected range.
+
+        Raises
+        ------
+        ValueError
+            If upper_bound < lower_bound or if actual does not lie outside the given interval.
+        """
+        if lower_bound is None and upper_bound is None:
+            raise ValueError("Illegal interval: Attempting to raise OutOfBoundsError, but no bounds given.")
+        # Use local variables with stricter types to help static analysis:
+        _lower_bound: Bound = lower_bound if lower_bound is not None else _MinInfinity()
+        _upper_bound: Bound = upper_bound if upper_bound is not None else _Infinity()
+        if _upper_bound.value < _lower_bound.value:
+            raise ValueError(
+                (
+                    f"Illegal interval: Attempting to raise OutOfBoundsError, but given upper bound {_upper_bound} is "
+                    f"actually less than given lower bound {_lower_bound}."
+                ),
+            )
+        elif _lower_bound.check_lower_bound(actual) and _upper_bound.check_upper_bound(actual):
+            raise ValueError(
+                (
+                    f"Illegal interval: Attempting to raise OutOfBoundsError, but value {actual} is not actually"
+                    f" outside given interval {_lower_bound.str_lower_bound()}, {_upper_bound.str_upper_bound()}."
+                ),
+            )
+        super().__init__(f"{actual} is not inside {_lower_bound.str_lower_bound()}, {_upper_bound.str_upper_bound()}.")
+
+
+class Bound(ABC):
+    """Abstract base class for (lower or upper) Bounds on a float value."""
+
+    def __init__(self, value: float):
+        self._value = value
+
+    def __str__(self) -> str:
+        """Get a string representation of the concrete value of the Bound."""
+        return str(self.value)
+
+    @abstractmethod
+    def str_lower_bound(self) -> str:
+        """Get a string representation of the Bound as the lower Bound of an interval."""
+
+    @abstractmethod
+    def str_upper_bound(self) -> str:
+        """Get a string representation of the Bound as the upper Bound of an interval."""
+
+    @abstractmethod
+    def check_lower_bound(self, value: float) -> bool:
```

Comment:
`value` parameter is not documented.

Also: `actual` would be a better name, so it matches the constructor of `OutOfBoundsError` and to avoid confusion with `self.value` (same for other places).

Reply:
Makes sense. Done.